tests/MongoDB.Driver.Core.Tests/Core/Connections/ConnectionInitializerTests.cs

```diff
@@ -39,6 +39,37 @@ public ConnectionInitializerTests()
             _subject = new ConnectionInitializer("test", new [] { new CompressorConfiguration(CompressorType.Zlib) });
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void InitializeConnection_should_acquire_connectionId_from_isMaster_response([Values(false, true)] bool async)
+        {
+            var isMasterReply = MessageHelper.BuildReply(
+                RawBsonDocumentHelper.FromJson("{ ok: 1, connectionId: 1 }"));
+            var buildInfoReply = MessageHelper.BuildReply(
+                RawBsonDocumentHelper.FromJson("{ ok: 1, version: \"4.2.0\" }"));
+            var gleReply = MessageHelper.BuildReply(
+                RawBsonDocumentHelper.FromJson("{ ok: 1, connectionId: 2 }"));
+
+            var connection = new MockConnection(__serverId);
+            connection.EnqueueReplyMessage(isMasterReply);
+            connection.EnqueueReplyMessage(buildInfoReply);
+            connection.EnqueueReplyMessage(gleReply);
```

Comment:
Does `gleReply` even need to be enqueued?

Reply:
Fixed.